syntax = "proto3";
import "protocol/site/server.proto";

enum DataType {
  NONE = 0;
  SNORT = 1;
  //추후 여러가지 데이터 버전들이 추가될것...
}

message DataVersionRequest {
  RequestServer server = 1;
}

message DataVersion{
  DataType type = 1;
  int64 version  = 2;
}

message DataVersionResponse{
  repeated DataVersion versions = 1;
}


// +1 하면서 버전 차이를 전부 팔로우한다.
message DataRequest{
  RequestServer server = 1;
  DataVersion version = 2;
}

message DataResponse{
  DataVersion version = 1;
  bytes data = 2;
}